solver/src/liquidity/offchain_orderbook.rs

```diff
@@ -28,6 +30,28 @@ struct OrderSettlementHandler {
     order: Order,
 }
 
+fn inflight_order_filter(order: Order, inflight_trades: &HashSet<Trade>) -> Option<Order> {
+    // TODO - could model inflight_trades as HashMap<OrderUid, Trade>
+    // https://github.com/gnosis/gp-v2-services/issues/673
+    // would avoid map-collect and make trade data accessible by id.
+    let inflight_trades_map = inflight_trades
```

Comment:
I'm not a huge fan of allocating a new map for every order being filtered just to do a more efficient search. You might as well just search linearly thought the `inflight_trades` as it would have the same runtime characteristic (`O(n)`) without having to allocate a map temporarily.

Reply:
I am totally fine with this approach. I was just trying to address Felix's comment (https://github.com/gnosis/gp-v2-services/pull/662#discussion_r642221566). Will revert to just a `HashSet<OrderUid>` for now then.

Reply:
I added a bunch of comments as to why I don't like `HashSet<Trade>`: https://github.com/gnosis/gp-v2-services/pull/662#discussion_r642390191